components/UserObserver.php

```diff
@@ -58,6 +67,11 @@ public function updateUser($params)
 
 					if (!$user) {
 						$user = new User();
+						$user_authentication = new UserAuthentication();
+						$user_authentication->password = $this->genPassword();
```

Comment:
out of curiosity, if I understand it correctly, this is generating a default password for the remote user and this should only run for the first time.

How would you let the user know the default password? Because after saving the $user_authentication, the password will be hashed.

Reply:
user is created as an inactive user in the db, but the password is required for saving. Normally user won't use this pass, it will be authenticated by ldap.

Reply:
right, understood, it is just a placeholder